src/fstab-generator/fstab-generator.c

```diff
@@ -1091,115 +1097,143 @@ static int sysroot_is_nfsroot(void) {
         }
 
         /* IPv4 address */
-        sep = strchr(arg_root_what, ':');
+        sep = strchr(what, ':');
         if (sep) {
-                a = strndupa_safe(arg_root_what, sep - arg_root_what);
+                a = strndupa_safe(what, sep - what);
 
                 if (in_addr_from_string(AF_INET, a, &u) >= 0)
                         return true;
         }
 
         /* root directory without address */
-        return path_is_absolute(arg_root_what) && !path_startswith(arg_root_what, "/dev");
+        return path_is_absolute(what) && !path_startswith(what, "/dev");
 }
 
-static int add_sysroot_mount(void) {
-        _cleanup_free_ char *what = NULL;
-        const char *opts, *fstype;
-        bool default_rw, makefs;
-        MountPointFlags flags;
+static bool validate_root_or_usr_mount_source(const char *what, const char *switch_name) {
         int r;
 
-        if (isempty(arg_root_what)) {
-                log_debug("Could not find a root= entry on the kernel command line.");
+        assert(switch_name);
+
+        if (isempty(what)) {
+                log_debug("Could not find a %s entry on the kernel command line.", switch_name);
```

Comment:
Reword this to be "switch_name not specified on ..." would be nice, now that more setups start to adopt gpt-auto